cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/DefaultStoreTest.java

```diff
@@ -190,13 +192,13 @@ public void testWorkflowNodeState() throws Exception {
     String mapReduceName = "mapReduce1";
     String sparkName = "spark1";
 
-    ProgramId mapReduceProgram = new ProgramId(namespaceName, appName, ProgramType.MAPREDUCE, mapReduceName);
-    ProgramId sparkProgram = new ProgramId(namespaceName, appName, ProgramType.SPARK, sparkName);
+    ProgramId mapReduceProgram = Ids.namespace(namespaceName).app(appName).mr(mapReduceName);
```

Comment:
nit: if you extract ``ApplicationId`` to a variable like ``Ids.namespace(namespaceName).app(appName)``, then lines 195, 196, 200 and probably more can be much shorter.